infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -336,32 +529,77 @@ private synchronized FileHandle getFileHandleForAddress(LogAddress logAddress) {
         return writeChannels.computeIfAbsent(filePath, a -> {
 
             try {
-                FileChannel fc = getChannel(a, false);
+                FileChannel fc1 = getChannel(a, false);
+                FileChannel fc2 = getChannel(getTrimmedFilePath(a) , false);
+                FileChannel fc3 = getChannel(getPendingTrimsFilePath(a), false);
 
                 boolean verify = true;
 
                 if (noVerify) {
                     verify = false;
                 }
 
-                writeHeader(fc, VERSION, verify);
+                writeHeader(fc1, VERSION, verify);
```

Comment:
The log will be generated using the version of this particular class. Any changes to the format will require an a new StreamLogFiles implementation with a different version